src/osd/PG.cc

```diff
@@ -3023,6 +3014,14 @@ void PG::init(
   write_if_dirty(*t);
 }
 
+void PG::shutdown()
+{
+  ch->flush();
```

Comment:
This call to ```flush()``` is one of those scary fixup patches that makes me wonder why it's necessary now but wasn't a problem before. Help?

Reply:
This is one of the races that I assumed was present in master too (since the earlier pg removal refactor)